### DISCUSSION TITLE
Ordnance shuttle type that can be retrieved as new

### DISCUSSION THREAD

**Discussion by Working-Joe** [FakeFishGames/Barotrauma#4909] [Ideas and suggestions]
While shuttles functioning as torpedoes, missiles and mines are fun to build, they cause trouble in campaigns as it is not always possible to retreive them. While shuttles can be retrieved at outposts, they will not be as new. This makes rearming the missiles hard or even impossible. 

It would be very nice to have a separate submarine class for these types of shuttles (ordnance?). When you then pay to retrieve a shuttle at outposts, if the shuttle is of the ordnance type it will be loaded in from the original file instead of loading the shuttle as it was abandoned.

Ideally the items on the shuttle would be taken into account in calculating a base price for the shuttle with some accuracy. This way people can't just create ordnance type shuttles with items to resupply them for cheap at each outpost. (Although balancing technically is the job of the sub builder)

**Comment by Xeavas** [comment]
An addition is to have a checkbox for it, and the ability to set the price yourself in the submarine editor.

**Reply by Psyduckdeluxe** [reply]
Torpedo fuctionality does seems as something a sub game SHOULD have.

**Reply by Xeavas** [reply]
_Why not torpedoes as a vanilla submarine weapon? :D_

**Comment by Leonidas-ai** [comment]
Yes!

**Comment by mLuby** [comment]
I agree, but instead of "Ordnance" I'd **prefer the more general "Drone" type**—distinct from subs and shuttles, which are crewed. The "Drone" shuttle type includes ordnance like torpedoes, mines, and gunbots, but also non-military use cases like disposable scouts, and resupply drones. This distinction would also help with [playing shuttles in campaign](https://github.com/Regalis11/Barotrauma/discussions/5136#discussioncomment-406515).

Alternatively, I'd support a "**create _every_ shuttle replacement anew**" behavior. If a shuttle is lost to the abyss, it makes more sense that you're buying a new one than that the old one is being miraculously recovered, along with whatever gear was inside. This might be easier to implement than separate Shuttle and Drone types.

Letting subwrights set the base replacement cost is also important.

My most wished-for shuttle feature is letting the player buy a _different_ shuttle or drone than the one they started with, but that's best left for another discussion.
